inference-engine/tests/functional/inference_engine/ngraph_reader/custom_op_tests.cpp

```diff
@@ -0,0 +1,241 @@
+            return InferenceEngine::StatusCode::OK;
+        }
+
+        InferenceEngine::StatusCode getSupportedConfigurations(std::vector<InferenceEngine::LayerConfig>& conf,
+                                                               InferenceEngine::ResponseDesc* /*resp*/) noexcept override {
+            InferenceEngine::LayerConfig layerConfig;
+            layerConfig.dynBatchSupport = true;
+
+            if (node->outputs().size() != 1 && node->inputs().size() != 1)
+                return InferenceEngine::GENERAL_ERROR;
+
+            InferenceEngine::DataConfig cfg;
+            cfg.constant = false;
+            cfg.inPlace = 0;
+
+            InferenceEngine::SizeVector order;
+            auto partialShape = node->get_output_partial_shape(0);
+            if (partialShape.is_dynamic())
+                return InferenceEngine::GENERAL_ERROR;
+
+            auto shape = node->get_output_shape(0);
+            for (size_t i = 0; i < shape.size(); i++) {
+                order.push_back(i);
+            }
+            cfg.desc = InferenceEngine::TensorDesc(InferenceEngine::Precision::I32,
+                                                   shape, {shape, order});
+            layerConfig.outConfs.push_back(cfg);
+            layerConfig.inConfs.push_back(cfg);
+            conf.push_back(layerConfig);
+            return InferenceEngine::OK;
+        }
+
+        InferenceEngine::StatusCode
+        execute(std::vector<InferenceEngine::Blob::Ptr>& inputs, std::vector<InferenceEngine::Blob::Ptr>& outputs,
+                InferenceEngine::ResponseDesc* /*resp*/) noexcept override {
+            for (size_t i = 0; i < inputs.size(); i++) {
+                InferenceEngine::MemoryBlob::CPtr minput = InferenceEngine::as<InferenceEngine::MemoryBlob>(inputs[i]);
+                InferenceEngine::MemoryBlob::Ptr moutput = InferenceEngine::as<InferenceEngine::MemoryBlob>(outputs[i]);
+                if (!moutput || !minput) {
+                    return InferenceEngine::StatusCode::PARAMETER_MISMATCH;
+                }
+                // locked memory holder should be alive all time while access to its buffer happens
+                auto minputHolder = minput->rmap();
+                auto moutputHolder = moutput->wmap();
+
+                // retrieve constant data
+                const auto data_beg = static_cast<char*>(data_ptr);
+                const auto data_end = std::next(data_beg, shape_size(shape) * el_type.size());
+                std::vector<char> value{data_beg, data_end};
+
+                auto inputData = minputHolder.as<const int32_t *>();
+                auto outputData = moutputHolder.as<int32_t  *>();
+                for (size_t j = 0; j < minput->size(); j++) {
+                    auto position = j * el_type.size() + 3;
+                    outputData[j] = inputData[j] + static_cast<int32_t>(value.at(position));
```

Comment:
Why do you need so complex code to get int32?
Looks like you can simple cast void* to int32_t* if you change the type of `data_ptr` from `void *` to `int32_t *`.
In this case you shouldn't calculate position and also you have an extra copy when you create `value` vector.

Reply:
Test refactored, no need to cast data to int32_t.